RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -0,0 +1,730 @@
+#include "CommonTools/CandAlgos/interface/ModifyObjectValueBase.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+
+#include "CondFormats/DataRecord/interface/GBRDWrapperRcd.h"
+#include "CondFormats/EgammaObjects/interface/GBRForestD.h"
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
+#include "CondFormats/EgammaObjects/interface/GBRForest.h"
+
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+
+#include "RecoEgamma/EgammaElectronAlgos/interface/RegressionHelper.h"
+
+#include "TFile.h"
+#include <vdt/vdtMath.h>
+
+namespace {
+  //const edm::EDGetTokenT<edm::ValueMap<float> > empty_token;
+  const edm::InputTag empty_tag;
+}
+
+#include <unordered_map>
+
+class EGExtraInfoModifierFromDB : public ModifyObjectValueBase {
+public:
+  typedef edm::EDGetTokenT<edm::ValueMap<float> > ValMapFloatToken;
+  typedef edm::EDGetTokenT<edm::ValueMap<int> > ValMapIntToken;
+
+  struct electron_config {
+    edm::InputTag electron_src;
+    edm::EDGetTokenT<edm::View<pat::Electron> > tok_electron_src;
+    edm::InputTag sigmaIetaIphi;
```

Comment:
Maybe define a ValueMapTagTokenPair struct and use it throughout.
Keeping these in sync otherwise appears tedious and error prone